source/utils/hasher.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <cstdint>
+
+#include "packet.hpp"
+
+namespace sim {
+
+class IHasher {
+public:
+    virtual ~IHasher() = default;
+    virtual std::uint32_t get_hash(Packet packet) const = 0;
+};
+
+class RandomHasher : public IHasher {
+public:
+    ~RandomHasher() = default;
+
+    std::uint32_t get_hash(Packet) const final;
```

Comment:
Just to keep the code style uniform
```suggestion
    std::uint32_t get_hash(Packet packet) const final;
```